packages/next/src/build/templates/pages-api.ts

```diff
@@ -24,3 +50,230 @@ export const routeModule = new PagesAPIRouteModule({
+  const { basePath, i18n, rewrites } = routesManifest
+
+  if (basePath) {
+    req.url = removePathPrefix(req.url || '/', basePath)
+  }
+
+  if (i18n) {
+    const urlParts = (req.url || '/').split('?')
+    const localeResult = normalizeLocalePath(urlParts[0] || '/', i18n.locales)
+
+    if (localeResult.detectedLocale) {
+      req.url = `${localeResult.pathname}${
+        urlParts[1] ? `?${urlParts[1]}` : ''
+      }`
+    }
+  }
+
+  const parsedUrl = parse(req.url || '/', true)
+  const pageIsDynamic = isDynamicRoute(srcPage)
+
+  const serverUtils = getUtils({
+    page: srcPage,
+    i18n,
+    basePath,
+    rewrites: Array.isArray(rewrites)
+      ? { beforeFiles: [], afterFiles: rewrites, fallback: [] }
+      : rewrites || {
+          beforeFiles: [],
+          afterFiles: [],
+          fallback: [],
+        },
+    pageIsDynamic,
+    trailingSlash: process.env.__NEXT_TRAILING_SLASH as any as boolean,
+    caseSensitive: Boolean(routesManifest.caseSensitive),
+  })
+  const rewriteParamKeys = Object.keys(
+    serverUtils.handleRewrites(req as any as BaseNextRequest, parsedUrl)
+  )
+  serverUtils.normalizeCdnUrl(req as any as BaseNextRequest, [
+    ...rewriteParamKeys,
+    ...Object.keys(serverUtils.defaultRouteRegex?.groups || {}),
+  ])
+
+  const params: Record<string, undefined | string | string[]> =
+    serverUtils.dynamicRouteMatcher
+      ? serverUtils.dynamicRouteMatcher(parsedUrl.pathname || '') || {}
+      : {}
+
+  const query = {
+    ...parsedUrl.query,
+    ...params,
+  }
+  serverUtils.normalizeQueryParams(query)
+
+  if (pageIsDynamic) {
+    const result = serverUtils.normalizeDynamicRouteParams(query, true)
+
+    if (result.hasValidParams) {
+      Object.assign(query, result.params)
+    }
+  }
+
+  // ensure instrumentation is registered and pass
+  // onRequestError below
+  const absoluteDistDir = path.join(dir, distDir)
+  await ensureInstrumentationRegistered(absoluteDistDir)
+
+  try {
+    const method = req.method || 'GET'
+    const tracer = getTracer()
+
+    const activeSpan = tracer.getActiveScopeSpan()
+
+    const invokeRouteModule = async (span?: Span) => {
+      await routeModule
+        .render(req, res, {
+          query,
+          params,
+          allowedRevalidateHeaderKeys: process.env
+            .__NEXT_ALLOWED_REVALIDATE_HEADERS as any as string[],
+          multiZoneDraftMode: Boolean(process.env.__NEXT_MULTI_ZONE_DRAFT_MODE),
+          trustHostHeader: process.env
+            .__NEXT_TRUST_HOST_HEADER as any as boolean,
+          // TODO: get this from from runtime env so manifest
+          // doesn't need to load
+          previewProps: prerenderManifest.preview,
+          propagateError: false,
+          dev: isDev,
+          page: 'VAR_DEFINITION_PAGE',
+
+          onError: (...args: Parameters<InstrumentationOnRequestError>) =>
+            instrumentationOnRequestError(absoluteDistDir, ...args),
+        })
+        .finally(() => {
```

Comment:
You're already using `await` in this block, maybe should have instead used a `try { } finally { }` block?